network/builder/src/builder.rs

```diff
@@ -43,75 +43,65 @@ use std::{
     sync::{Arc, RwLock},
 };
 use subscription_service::ReconfigSubscription;
-use tokio::runtime::{Builder, Handle, Runtime};
+use tokio::runtime::Handle;
+
+#[derive(Debug, PartialEq, PartialOrd)]
+enum State {
+    CREATED,
+    BUILT,
+    STARTED,
+}
 
 /// Build Network module with custom configuration values.
 /// Methods can be chained in order to set the configuration values.
 /// MempoolNetworkHandler and ConsensusNetworkHandler are constructed by calling
 /// [`NetworkBuilder::build`].  New instances of `NetworkBuilder` are obtained
-/// via [`NetworkBuilder::new`].
-// TODO(philiphayes): refactor NetworkBuilder and libra-node; current config is
-// pretty tangled.
+/// via [`NetworkBuilder::create`].
 pub struct NetworkBuilder {
-    executor: Handle,
+    state: State,
+    executor: Option<Handle>,
     network_context: Arc<NetworkContext>,
-    seed_peers: HashMap<PeerId, Vec<NetworkAddress>>,
-    trusted_peers: Arc<RwLock<HashMap<PeerId, x25519::PublicKey>>>,
-    channel_size: usize,
-    connectivity_check_interval_ms: u64,
-    max_connection_delay_ms: u64,
-    /// For now full node connections are limited by
-    max_fullnode_connections: usize,
 
     configuration_change_listener_builder: Option<ConfigurationChangeListenerBuilder>,
     connectivity_manager_builder: Option<ConnectivityManagerBuilder>,
     discovery_builder: Option<DiscoveryBuilder>,
     health_checker_builder: Option<HealthCheckerBuilder>,
     peer_manager_builder: PeerManagerBuilder,
 
+    // (StateSync) ReconfigSubscriptions required by internal Network components.
     reconfig_subscriptions: Vec<ReconfigSubscription>,
 }
 
 impl NetworkBuilder {
     /// Return a new NetworkBuilder initialized with default configuration values.
+    // TODO:  Remove `pub`.  NetworkBuilder should only be created thorugh `::create()`
     pub fn new(
-        executor: Handle,
         chain_id: ChainId,
-        network_id: NetworkId,
-        role: RoleType,
-        peer_id: PeerId,
+        trusted_peers: Arc<RwLock<HashMap<PeerId, x25519::PublicKey>>>,
+        network_context: Arc<NetworkContext>,
         listen_address: NetworkAddress,
         authentication_mode: AuthenticationMode,
     ) -> NetworkBuilder {
```

Comment:
nit -> Self when inside a an impl for a struct

Reply:
done.